### ISSUE TITLE
Enhancement to sort-keys to optionally sort destructured function parameters

### ISSUE THREAD

**Issue by IRoninCoder** [eslint/eslint#14873]
**What rule do you want to change?**
sort-keys
**Does this change cause the rule to produce more or fewer warnings?**
more warnings
**How will the change be implemented? (New option, new default behavior, etc.)?**
New option
**Please provide some example code that this change will affect:**

```js
var First = ({ one, two, three }) => { return "hello world"; };
// functionParams is the new option I've added in my branch
options: ["desc", { caseSensitive: false, functionParams: true }]
```

**What does the rule currently do for this code?**
It breaks with following error when running unit tests:
```js
var First = ({ one, two, three }) => { return "hello world"; };
// AssertionError [ERR_ASSERTION]: A fatal parsing error occurred: Parsing error: Unexpected token ,
```
**What will the rule do after it's changed?**
It'll optionally enable to sort destructured parameter keys to be sorted in the same way object literal keys are sorted currently.
**Are you willing to submit a pull request to implement this change?**
Yes, but I have questions:
1. What is causing the parsing failure?
2. Is it possible to fix the parsing failure while maintaining backward compatibility for existing `eslint` consumers?

**Comment by IRoninCoder** [comment]
related https://github.com/yannickcr/eslint-plugin-react/pull/2854

**Comment by mdjermanovic** [comment]
Hi @IRoninCoder, thanks for the issue!

ESLint's core stylistic rules are [frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we won't be adding any more options to the `sort-keys` rule. You can always make a custom rule.

I'm closing this issue per our rules policies.

> What is causing the parsing failure?

Default `ecmaVersion` is 5. If you specify `parserOptions: { ecmaVersion: 6 } }` in the test case, it should work well.
